util/unifyfs/src/unifyfs-rm.c

```diff
@@ -483,58 +485,120 @@ static int slurm_read_resource(unifyfs_resource_t* resource)
+	    argc += 2;
+	}
 
-    if (args->consistency != UNIFYFS_CM_LAMINATED) {
+	if (args->cleanup) {
+            if (server_argv != NULL) {
+	        server_argv[argc] = strdup("-C");
+            }
+	    argc++;
+	}
+
+	if (args->consistency != UNIFYFS_CM_LAMINATED) {
+            if (server_argv != NULL) {
+	        server_argv[argc] = strdup("-c");
+		server_argv[argc + 1] =
+		    strdup(unifyfs_cm_enum_str(args->consistency));
+            }
+	    argc += 2;
+	}
+
+	if (args->mountpoint != NULL) {
+            if (server_argv != NULL) {
+	        server_argv[argc] = strdup("-m");
+                server_argv[argc + 1] = strdup(args->mountpoint);
+            }
+	    argc += 2;
+	}
+
+	if (server_argv != NULL) {
+	    server_argv[argc] = strdup("-S");
+	    server_argv[argc + 1] = strdup(args->share_dir);
+	    server_argv[argc + 2] = strdup("-H");
+	    server_argv[argc + 3] = strdup(args->share_hostfile);
+	    argc += 4;
+	}
+	break;
+    case UNIFYFS_CONSTRUCT_ARGS_STAGE_IN:
         if (server_argv != NULL) {
-            server_argv[argc] = strdup("-c");
-            server_argv[argc + 1] =
-                strdup(unifyfs_cm_enum_str(args->consistency));
-        }
-        argc += 2;
-    }
+            if (args->transfer_exe == NULL) {
+	        fprintf(stderr,
+			"%s: args->transfer_exe is NULL!\n",
+			__func__);
+                return -1;
+            }
+            if (args->stage_in == NULL) {
+                fprintf(stderr, "%s: args->stage_in is NULL!\n",
+			__func__);
+	        return -1;
+            }
 
-    if (args->mountpoint != NULL) {
+	    //	    if (args->server_path != NULL) {
+	    server_argv[0] = strdup(args->transfer_exe);
+	    //    } else {
+	    //	        server_argv[0] = strdup(BINDIR "/unifyfsd");
+	    //	    }
+	    argc = 1;
+
+	    // if we need to add arguments betwen the command and the
+	    // manifest file name they get added here.
+
+	    server_argv[argc] = strdup(args->stage_in);
+	    argc++;
+	}
+	break;
+    case UNIFYFS_CONSTRUCT_ARGS_STAGE_OUT:
         if (server_argv != NULL) {
-            server_argv[argc] = strdup("-m");
-            server_argv[argc + 1] = strdup(args->mountpoint);
-        }
-        argc += 2;
-    }
+            if (args->transfer_exe == NULL) {
+	        fprintf(stderr, "%s: args->transfer_exe is NULL!\n",
+			__func__);
+	        return -1;
+            }
+            if (args->stage_in == NULL) {
+	        fprintf(stderr, "%s: args->stage_out is NULL!\n",
+			__func__);
+	        return -1;
+            }
 
-    if (server_argv != NULL) {
-        server_argv[argc] = strdup("-S");
-        server_argv[argc + 1] = strdup(args->share_dir);
-        server_argv[argc + 2] = strdup("-H");
-        server_argv[argc + 3] = strdup(args->share_hostfile);
+	    //	    if (args->server_path != NULL) {
+	    server_argv[0] = strdup(args->transfer_exe);
+	    //    } else {
+	    //	        server_argv[0] = strdup(BINDIR "/unifyfsd");
+	    //	    }
+	    argc = 1;
```

Comment:
Remove debug comments

Reply:
Explanation in the other thread.  Same reason.